matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/identity/DefaultIdentityService.kt

```diff
@@ -0,0 +1,345 @@
+            callback.onFailure(IdentityServiceError.OutdatedHomeServer)
+            return NoOpCancellable
+        }
+
+        return GlobalScope.launchToCallback(coroutineDispatchers.main, callback) {
+            bindThreePidsTask.execute(BindThreePidsTask.Params(threePid))
+        }
+    }
+
+    override fun submitValidationToken(threePid: ThreePid, code: String, callback: MatrixCallback<Unit>): Cancelable {
+        return GlobalScope.launchToCallback(coroutineDispatchers.main, callback) {
+            submitTokenForBindingTask.execute(IdentitySubmitTokenForBindingTask.Params(threePid, code))
+        }
+    }
+
+    override fun unbindThreePid(threePid: ThreePid, callback: MatrixCallback<Unit>): Cancelable {
+        if (homeServerCapabilitiesService.getHomeServerCapabilities().lastVersionIdentityServerSupported.not()) {
+            callback.onFailure(IdentityServiceError.OutdatedHomeServer)
+            return NoOpCancellable
+        }
+
+        return GlobalScope.launchToCallback(coroutineDispatchers.main, callback) {
+            unbindThreePidsTask.execute(UnbindThreePidsTask.Params(threePid))
+        }
+    }
+
+    override fun isValidIdentityServer(url: String, callback: MatrixCallback<Unit>): Cancelable {
+        return GlobalScope.launchToCallback(coroutineDispatchers.main, callback) {
+            val api = retrofitFactory.create(unauthenticatedOkHttpClient, url).create(IdentityAuthAPI::class.java)
+
+            identityPingTask.execute(IdentityPingTask.Params(api))
+        }
+    }
+
+    override fun disconnect(callback: MatrixCallback<Unit>): Cancelable {
+        return GlobalScope.launchToCallback(coroutineDispatchers.main, callback) {
+            identityDisconnectTask.execute(Unit)
+
+            identityStore.setUrl(null)
+            updateIdentityAPI(null)
+            updateAccountData(null)
+        }
+    }
+
+    override fun setNewIdentityServer(url: String, callback: MatrixCallback<String>): Cancelable {
+        val urlCandidate = buildString {
+            if (!url.startsWith("http")) {
+                append("https://")
+            }
+            append(url)
+        }
+
+        return GlobalScope.launchToCallback(coroutineDispatchers.main, callback) {
+            val current = getCurrentIdentityServerUrl()
+            if (urlCandidate == current) {
+                // Nothing to do
+                Timber.d("Same URL, nothing to do")
+            } else {
+                // Disconnect previous one if any, first, because the token will change.
+                // In case of error when configuring the new identity server, this is not a big deal,
+                // we will ask for a new token on the previous Identity server
+                runCatching {
+                    identityDisconnectTask.execute(Unit)
+                }
+
+                // Try to get a token
+                val token = getNewIdentityServerToken(urlCandidate)
```

Comment:
Could throw -> broke Global scope and cascade to all child scopes?

Reply:
no